nixos/modules/config/sysctl.nix

```
@@ -21,19 +21,27 @@ in
   options = {
 
     boot.kernel.sysctl = mkOption {
-      type = types.submodule {
+      type = let
+        highestValueType = types.nullOr types.ints.unsigned // {
```

It may be better to factor out `types.nullOr`.

```suggestion
        highestValueType = types.ints.unsigned // {
```